crates/typst-cli/src/compile.rs

```diff
@@ -78,8 +78,28 @@ pub fn compile_once(
         Status::Compiling.print(command).unwrap();
     }
 
-    // Ensure that the main file is present.
-    world.source(world.main()).map_err(|err| err.to_string())?;
+    // Check the encoding of main file.
+    if let Err(typst::diag::FileError::InvalidUtf8) = world.source(world.main()) {
```

Comment:
the `compile` call below expects the source file to exist, so if this fails for another reason than `InvalidUtf8`, it will likely crash. is there a particular reason to handle this specific error here, rather than just converting any `FileResult` to a `SourceResult` with the `at(Span::detached())` conversion and printing the error directly?